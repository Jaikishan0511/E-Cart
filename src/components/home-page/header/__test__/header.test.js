import React from "react";
import Header from "../index";
import ReactTestRenderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
const HeaderComp = (
    <Router>
      <Header />
    </Router>
  );
describe("Actions", () => {
    describe("Render", () => {
      it("should be a valid React Element", () => {
        expect(React.isValidElement(<Header />)).toBeTruthy();
      });
  
      it("should render without throwing an error", () => {
        expect(() =>
          ReactTestRenderer.create(HeaderComp)
        ).not.toThrow();
      });
    });
  
    describe("Snapshots", () => {
      it("matches the default snapshot", () => {
        const tree = ReactTestRenderer.create(HeaderComp).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });