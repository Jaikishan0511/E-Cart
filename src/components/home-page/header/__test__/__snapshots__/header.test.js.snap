// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions Snapshots matches the default snapshot 1`] = `
<div
  className="header-container"
>
  <div
    className="title"
  >
     MyCart 
  </div>
  <div
    className="search-bar"
  >
    <img
      alt="logo"
      src="/images/search-icon.png"
    />
    <input
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="cart-img"
  >
    <a
      href="/view-cart"
      onClick={[Function]}
    >
      <img
        alt="logo"
        src="/images/cart.png"
      />
      <span
        className="badge"
      />
    </a>
  </div>
  <div />
</div>
`;
